huaweicloud/services/aom/resource_huaweicloud_aom_prom_instance.go

```diff
@@ -0,0 +1,214 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourcePromInstanceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	product := "aom"
+
+	client, err := cfg.NewServiceClient(product, cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating AOM client: %s", err)
+	}
+
+	createPrometheusInstanceHttpUrl := "v1/{project_id}/aom/prometheus"
+	createPrometheusInstanceHttpUrl = strings.ReplaceAll(createPrometheusInstanceHttpUrl, "{project_id}", client.ProjectID)
+	createPrometheusInstancePath := client.Endpoint + createPrometheusInstanceHttpUrl
+
+	createPrometheusInstanceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	createPrometheusInstanceOpt.JSONBody = utils.RemoveNil(buildCreatePrometheusInstanceBodyParams(d, cfg))
+	createPrometheusInstanceResp, err := client.Request("POST", createPrometheusInstancePath, &createPrometheusInstanceOpt)
+	if err != nil {
+		return diag.Errorf("error creating AOM prometheus instance: %s", err)
+	}
+	createPrometheusInstanceRespBody, err := utils.FlattenResponse(createPrometheusInstanceResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("prometheus[0].prom_id", createPrometheusInstanceRespBody)
+	if err != nil {
+		return diag.Errorf("error creating AOM prometheus instance: ID is not found in API response")
+	}
+
+	d.SetId(id.(string))
+
+	return resourcePromInstanceRead(ctx, d, meta)
+}
+
+func buildCreatePrometheusInstanceBodyParams(d *schema.ResourceData, cfg *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"prom_name":             d.Get("prom_name"),
+		"prom_type":             d.Get("prom_type"),
+		"enterprise_project_id": utils.ValueIngoreEmpty(common.GetEnterpriseProjectID(d, cfg)),
+	}
+	return bodyParams
+}
+
+func resourcePromInstanceRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	product := "aom"
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating AOM client: %s", err)
+	}
+
+	getPrometheusInstanceHttpUrl := "v1/{project_id}/aom/prometheus"
+	getPrometheusInstanceHttpUrl = strings.ReplaceAll(getPrometheusInstanceHttpUrl, "{project_id}", client.ProjectID)
+	getPrometheusInstanceHttpUrl += fmt.Sprintf("?prom_id=%s", d.Id())
+	getPrometheusInstancePath := client.Endpoint + getPrometheusInstanceHttpUrl
+
+	getPrometheusInstanceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+	getPrometheusInstanceResp, err := client.Request("GET", getPrometheusInstancePath, &getPrometheusInstanceOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving AOM prometheus instance")
```

Comment:
```suggestion
		return diag.Errorf(""error retrieving AOM prometheus instance: %s", err)
```
This method will not return a 404 status code.